apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl.erl

```diff
@@ -24,10 +24,11 @@ hosts(Hosts) when is_list(Hosts) ->
     kpro:parse_endpoints(Hosts).
 
 %% Client ID is better to be unique to make it easier for Kafka side trouble shooting.
-make_client_id(KafkaType0, BridgeName0) ->
-    KafkaType = to_bin(KafkaType0),
-    BridgeName = to_bin(BridgeName0),
-    iolist_to_binary([KafkaType, ":", BridgeName, ":", atom_to_list(node())]).
+%% Do not use Resource ID as it'll collide when a Kafka Bridge and an Azure Event Hub
+%% Bridge have the same name (as both resources will be <kafka type>:<bridge name>).
+make_client_id(InstanceId) ->
+    InstanceIdBin = to_bin(InstanceId),
```

Comment:
We're actually using the resource ID with this change.

Another thing that I've just realized: this actually changes the clientid for existing bridges.  For consumers, in particular, it means that a rolling upgrade might miss some messages, as it'll have a new offset in Kafka.

Perhaps we could just ensure that the `bridge:` prefix is removed from the resource id to avoid such a situation.

```suggestion
make_client_id(InstanceId) ->
    InstanceIdBin0 = to_bin(InstanceId),
    InstanceIdBin = binary:replace(InstanceIdBin0, <<"bridge:">>, <<>>),
```

Reply:
Good catch! =)